cmd/kubeadm/app/cmd/phases/addons.go

```diff
@@ -145,6 +145,7 @@ func getAddonsSubCommands() []*cobra.Command {
 
 		if properties.use == "all" || properties.use == "kube-proxy" {
 			cmd.Flags().StringVar(&cfg.API.AdvertiseAddress, "apiserver-advertise-address", cfg.API.AdvertiseAddress, `The IP address or DNS name the API server is accessible on`)
+			cmd.Flags().StringVar(&cfg.API.AdvertiseDNSAddress, "apiserver-advertise-dns-address", cfg.API.AdvertiseDNSAddress, `The DNS name the API server is accessible on`)
```

Comment:
I think I've turned a blind eye to some of the flag proliferation, this seems awfully redundant.  

/cc @errordeveloper

Reply:
part of the issue is the previous advertiseaddress was actually a listen address not an advertise.